### PR TITLE
Add different food types. Add tests for some new functions

### LINKED ISSUE #1: Need to add food types
Food types:
    * Neutral (white)
    * Aggressive (red)
    * Valuable (color similar to snake color)

### PR DESCRIPTION
**Pull request by aloneik**
* Resolve https://github.com/aloneik/custom_snake/issues/1
* Add logic to update game score for different food types
* Add test for `updateGameScore` function and `Food::setType` method
* Add new lines at end of files

### CODE REVIEW

File: src/food.cpp [hunk 1]
```diff
@@ -1,12 +1,46 @@
...
+    _shape.setFillColor(sf::Color::White);
+}
+
+Food::Food(const sf::Vector2f position, const sf::Vector2f size, const FoodType type) :
+    _type(FoodType::Neutral), _shape(size)
 {
     _shape.setPosition(position);
-    _shape.setFillColor(_color);
+    _shape.setFillColor(sf::Color::White);
 }
```
Comment (culhatsker) [inline src/food.cpp]: Too many constructors. Do you actually use them? Even if you do, I doubt it's necessary.

File: src/food.cpp [hunk 2]
```diff
@@ -1,12 +1,46 @@
...
+        case FoodType::Aggressive:
+            _shape.setFillColor(sf::Color::Red);
+            break;
+        case FoodType::Valuable:
+            _shape.setFillColor(sf::Color::Magenta);
+            break;
+        default:
+            std::cout << "Unknown food type!!" << std::endl;
+    }
+}
```
Comment (culhatsker) [inline src/food.cpp:34]: Do you need to change the type of the food after creating it? If no, then you don't need this function, move it to the constructor.

File: src/include/food.hpp [hunk 3]
```diff
@@ -5,25 +5,39 @@
...
+};
+
+
 class Food : public sf::Drawable {
   public:
-    explicit Food() = default;
+    explicit Food();
 
-    explicit Food(const sf::Vector2f position);
+    explicit Food(const sf::Vector2f position, const sf::Vector2f size);
```
Comment (culhatsker) [inline src/include/food.hpp]: AFAIK, there's no need in `explicit` if the constuctor takes more than one argument.

File: src/include/utilites.hpp [hunk 4]
```diff
@@ -60,6 +60,46 @@ bool isIntersectVector2VectorArray(const ShapeT& shape, const VectorShapeT& shap
 
 inline void updateGameScore(std::size_t* gameScore, const Food& food)
```
Comment (culhatsker) [inline src/include/utilites.hpp:61]: Can you pass the game score by reference, not by pointer?

File: src/include/utilites.hpp [hunk 5]
```diff
@@ -60,6 +60,46 @@ bool isIntersectVector2VectorArray(const ShapeT& shape, const VectorShapeT& shap
...
+    }
 }
+
+sf::Vector2f generatePosition(const sf::Rect<std::size_t>& bounds)
+{
+    static std::random_device seeder;
+    static std::mt19937 rng{seeder()};
+   
+    std::cout << "Bounds:" << std::endl;
+    std::cout << bounds.left << " " << bounds.top << " " <<  bounds.left + bounds.width << " " << bounds.top + bounds.height << std::endl;
```
Comment (culhatsker) [inline src/include/utilites.hpp]: Debug prints?

File: src/main.cpp [hunk 6]
```diff
@@ -93,20 +98,29 @@ int main()
...
-                    std::random_device seeder;
-                    std::mt19937 rng(seeder());
-                    std::uniform_int_distribution<> xDistr(0u, (GAME_FIELD_WIDTH / BLOCK_SIZE) - 1u);
-                    std::uniform_int_distribution<> yDistr(0u, (GAME_FIELD_HEIGHT / BLOCK_SIZE) - 1u);
-                    float xCoord{xDistr(rng) * static_cast<float>(BLOCK_SIZE)};
-                    float yCoord{yDistr(rng) * static_cast<float>(BLOCK_SIZE)};
-                    food.setPosition(sf::Vector2f{xCoord, yCoord});
+                    sf::Vector2f newFoodPosition = generatePosition(gameFieldBounds) * static_cast<float>(BLOCK_SIZE);
+                    food.setPosition(newFoodPosition);
                     while (isIntersectVector2VectorArray(food, getSegments(snake)))
```
Comment (culhatsker) [inline src/main.cpp:104]: What does this block do and why does it copy the previous two lines?

File: src/main.cpp [hunk 7]
```diff
@@ -93,20 +98,29 @@ int main()
...
+                        food.setType(FoodType::Valuable);
+                    }
+                    else if ((pseudoRandomNumber % 2) == 0)
+                    {
+                        food.setType(FoodType::Aggressive);
+                    }
+                    else 
+                    {
+                        food.setType(FoodType::Neutral);
+                    }
```
Comment (culhatsker) [inline src/main.cpp:124]: Why don't you use the same generator that you used before. The frame time in integer milliseconds is not very random, it usually equals to `1000/60`

**Review by culhatsker — CHANGES_REQUESTED** [review]

File: src/food.cpp (hunk 1 above)
Comment (aloneik) [reply src/food.cpp]: Thanks. I think what third constructor is enough

File: src/include/food.hpp (hunk 3 above)
Comment (aloneik) [reply src/include/food.hpp]: Yes, thanks.

File: src/include/utilites.hpp (hunk 4 above)
Comment (aloneik) [reply src/include/utilites.hpp:61]: It's recommended to pass arguments, which will be changed, by pointer.

File: src/main.cpp (hunk 7 above)
Comment (aloneik) [reply src/main.cpp:124]: There will be special logic to choose food type

File: src/food.cpp (hunk 2 above)
Comment (aloneik) [reply src/food.cpp:34]: Now it is need, because i create only one food object.

**Pull request merged**